exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/statistics/AvgWidthMergedStatistic.java

```diff
@@ -0,0 +1,135 @@
+
+  public AvgWidthMergedStatistic (String name, String inputName) {
+    this.name = name;
+    this.inputName = inputName;
+    this.sumHolder = new HashMap<>();
+    types = nonNullStatCounts = statCounts = null;
+  }
+
+  @Override
+  public String getName() {
+    return name;
+  }
+
+  @Override
+  public String getInput() {
+    return inputName;
+  }
+
+  @Override
+  public void merge(ValueVector input) {
+    // Check the input is a Map Vector
+    assert (input.getField().getType().getMinorType() == TypeProtos.MinorType.MAP);
+    MapVector inputMap = (MapVector) input;
+    for (ValueVector vv : inputMap) {
+      String colName = vv.getField().getLastName();
+      NullableFloat8Holder colSumHolder;
+      if (sumHolder.get(colName) != null) {
+        colSumHolder = (NullableFloat8Holder) sumHolder.get(colName);
+      } else {
+        colSumHolder = new NullableFloat8Holder();
+        sumHolder.put(colName, colSumHolder);
+      }
+      Object val = vv.getAccessor().getObject(0);
+      if (val != null) {
+        colSumHolder.value += (double) val;
+        colSumHolder.isSet = 1;
+      }
+    }
+  }
+
+  @Override
+  public Object getStat(String colName) {
+      if (mergeComplete != true) {
+        throw new IllegalStateException(
+            String.format("Statistic `%s` has not completed merging statistics", name));
+      }
+      NullableFloat8Holder colSumHolder = (NullableFloat8Holder) sumHolder.get(colName);
+      return (long) (colSumHolder.value/ getRowCount(colName));
+    }
+
+  @Override
+  public void setOutput(ValueVector output) {
+    // Check the input is a Map Vector
+    assert (output.getField().getType().getMinorType() == TypeProtos.MinorType.MAP);
+    // Dependencies have been configured correctly
+    assert (configureComplete == true);
+    MapVector outputMap = (MapVector) output;
+
+    for (ValueVector outMapCol : outputMap) {
+      String colName = outMapCol.getField().getLastName();
+      NullableFloat8Holder colSumHolder = (NullableFloat8Holder) sumHolder.get(colName);
+      NullableFloat8Vector vv = (NullableFloat8Vector) outMapCol;
+      vv.allocateNewSafe();
+      vv.getMutator().setSafe(0, (colSumHolder.value / getRowCount(colName)));
+    }
+    mergeComplete = true;
+  }
+
+  @Override
+  public void configure(Object configurations) {
+    List<MergedStatistic> statistics = (List<MergedStatistic>) configurations;
+    for (MergedStatistic statistic : statistics) {
+      if (statistic.getName().equals("type")) {
+        types = statistic;
+      } else if (statistic.getName().equals("statcount")) {
+        statCounts = statistic;
+      } else if (statistic.getName().equals("nonnullstatcount")) {
+        nonNullStatCounts = statistic;
+      }
+    }
+    assert (types != null && statCounts != null && nonNullStatCounts != null);
+    configureComplete = true;
+  }
+
+  private long getRowCount(String colName) {
+    int type = (int) types.getStat(colName);
```

Comment:
Please use the enum value here: MinorType (or TypeProtos.MinorType, but Java will import the nested type for you so the code is simpler.)

Java allows code of the form:

```
if (type == MinorType.VAR16CHAR || type == ... ) {
```

or, more simply:

```
switch(type) {
case VAR16CHAR:
case VARCHAR:
case VARBINARY:
  return ...
default:
  return ...
}
```